openeogeotrellis/backend.py

```diff
@@ -2024,6 +2024,11 @@ def as_boolean_arg(job_option_key: str, default_value: str) -> str:
 
             batch_job_cfg_secret_name = k8s_get_batch_job_cfg_secret_name(spark_app_id)
 
+            max_result_size = '5g'
+            from pyspark.util import _parse_memory
+            if _parse_memory(max_result_size) < _parse_memory(options.executor_memory):
+                max_result_size = options.executor_memory
```

Comment:
I believe you'll want to check against driver_memory, the max_result_size is what gets collected into the driver, so the combination of all the data in the partitions stored on executors.